plugins/polio/budget/serializers.py

```diff
@@ -129,60 +153,65 @@ class Meta:
             "created_at": {"read_only": True},
         }
 
-    def validate_rounds(self, submitted_rounds: list[Round]) -> list[Round]:
+    def validate_rounds(self, submitted_rounds: list[OrderedDict]) -> list[OrderedDict]:
         request = self.context["request"]
         is_new = self.instance is None
+        round_ids = [round_dict.get("id") for round_dict in submitted_rounds]
+
+        rounds = Round.objects.filter(id__in=round_ids).select_related("campaign", "budget_process")
 
         valid_rounds_ids = Campaign.objects.filter_for_user(request.user).values_list("rounds", flat=True)
-        invalid_rounds = [round for round in submitted_rounds if round.id not in valid_rounds_ids]
-        if invalid_rounds:
-            raise serializers.ValidationError(f"The user does not have the permissions for rounds: {invalid_rounds}.")
+        invalid_round_ids = [round.id for round in rounds if round.id not in valid_rounds_ids]
+        if invalid_round_ids:
+            raise serializers.ValidationError(
+                f"The user does not have the permissions for rounds: {invalid_round_ids}."
+            )
 
         if is_new:
-            already_linked_rounds = [round for round in submitted_rounds if round.budget_process]
-            if already_linked_rounds:
+            already_linked_round_ids = [round.id for round in rounds if round.budget_process]
+            if already_linked_round_ids:
                 raise serializers.ValidationError(
-                    f"A BudgetProcess already exists for rounds: {already_linked_rounds}."
+                    f"A BudgetProcess already exists for rounds: {already_linked_round_ids}."
                 )
 
-        rounds_campaigns = {round.campaign_id for round in submitted_rounds}
+        rounds_campaigns = {round.campaign_id for round in rounds}
         if len(rounds_campaigns) > 1:
-            raise serializers.ValidationError(f"Rounds must be from the same campaign: {submitted_rounds}.")
+            raise serializers.ValidationError("Rounds must be from the same campaign.")
 
         return submitted_rounds
 
-    def create(self, validated_data: dict) -> BudgetProcess:
-        request = self.context["request"]
+    def handle_rounds(self, budget_process, rounds_data):
```

Comment:
```suggestion
    def handle_rounds(self, budget_process: BudgetProcess, rounds_data: dict) -> None:
```

Let's be consistent :)